client/src/components/tabs/Landing.jsx

```diff
@@ -0,0 +1,200 @@
+import React from "react";
+import "../../styles/Landing.css";
+import configuration from "../../config/constants.js";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import { useState,useEffect } from "react";
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import axios from 'axios';
+import Cookies from 'js-cookie';
+import { addProf } from '../../redux/actions/auth';
+import CreateAccount from '../forms/CreateAccount';
+
+export const Landing = ({ isAuthenticated,profAuthenticated,addProf }) => {
+
+  let token = Cookies.get("token") ? Cookies.get("token") : null;
+  
+  addProf(token);
+
+  const [open, setOpen] = useState({
+    isOpen: false,
+    username: "",
+    password: "",
+    incorrectUsername: false,
+    incorrectPassword: false,
+    profAuthenticated: false,
+    isCreate: false,
+  });
+  const handleCreate = () => {
+    setOpen({
+      ...open,
+      isCreate: !open.isCreate
+    });
+  }
+  const handleClickOpen = () => {
+    setOpen({ ...open, isOpen: true });
+  };
+
+  const handleClose = () => {
+    setOpen({ ...open, isOpen: false });
+  };
+
+  const editUsername = (e) => {
+    e.preventDefault();
+    setOpen({
+      ...open,
+      username: e.target.value,
+      incorrectUsername: false
+    });
+  }
+  const editPassword = (e) => {
+    e.preventDefault();
+    setOpen({
+      ...open,
+      password: e.target.value,
+      incorrectPassword: false,
+    });
+  }
+  const config = {
+    headers: {
+      "Content-Type": "application/json",
+      "Access-Control-Allow-Origin": "*",
+
+    }
+  };
+  const submit = async (e) => {
+    let iu = false, ip = false, pa = false;
+    e.preventDefault();
+    await axios.post("/api/ProfAuth",
+      {
+        username: open.username,
+        password: open.password
+      }, config).then(function (res) {
+        
+        if (!res || res.status === 204) {
+          iu = true;
+        }
+        else {
+          iu = false
+          if (res.status === 206) { ip = true; }
+          else {
+            ip = false;
+            pa = true;
+            //store the token in HTTP cookie
+            Cookies.set('token', res.data.token, { expires: 1 });
+            window.alert("welcome");
+            addProf(res.data.token);
+          }
+        }
+        setOpen({
+          ...open,
+          incorrectUsername: iu,
+          incorrectPassword: ip,
+          profAuthenticated: pa,
+        });
```

Comment:
Since you're using redux anyways, maybe its possible to shift this to a redux action as well?

Reply:
Well this is done only for the login page
pointless to keeping this as a global state

Reply:
Or do you mean to store the cookie in the actions part?

Reply:
Not the cookies. The whole thing. 
Invalid credentials are more of an event than a state so something like dispatching an action that triggers an alert would be a better way to go, but that can be low-priority as of now ig.